### ISSUE TITLE
Use schema:DefinedTerm for SEO

### ISSUE THREAD

**Issue by gkellogg** [w3c/respec#1625]
schema.org is introducing [DefinedTerm](https://github.com/schemaorg/schemaorg/pull/1776), which may be useful to add to the JSON-LD output in SEO.

TermDefinitions are contained within a DefinedTermSet, which would use to reveal terms defined within a spec using `dfn` as part of the JSON-LD blob. For example:

```json
{
  "@context": [
    "http://schema.org",
    {
      "@vocab": "http://schema.org/",
      "@language": "en",
      "w3p": "http://www.w3.org/2001/02pd/rec54#",
      "foaf": "http://xmlns.com/foaf/0.1/",
      "datePublished": {"@type": "xsd:date"},
      "inLanguage": {"@language": null},
      "isBasedOn": {"@type": "@id"},
      "license": {"@type": "@id"}
    }
  ],
  "id": "https://w3c.github.io/some-API/",
  "type": ["TechArticle", "TermDefinitionSet"],
  "name": "Replace me with a real title",
  ...
  "hasDefinedTerm": [{
    "@id": "https://w3c.github.io/some-API/#dfn-Foo",
    "@type": "DefinedTerm",
    "name": "Foo",
    "description": "The Foo interface is nice. Lets you do stuff."
  }]
}

```

**Comment by marcoscaceres** [comment]
@gkellogg, seems interesting. But `description` seems kinda heavy headed... would duplicate a lot of content that is that https://w3c.github.io/some-API/#dfn-Foo. Maybe that's ok.

**Comment by marcoscaceres** [comment]
@gkellogg, as before, extra bonus points 🌟 if any service is exposing this to users.

**Comment by gkellogg** [comment]
Given that it’s still a proposed term, I wouldn’t expect to see any use in the field yet.  Once it’s in, adding this would be straightforward, probably controlled by yet another flag.

Term definitions are some of the most useful enrypoints into a spec, IMHO, and providing more SEO support for them could be a good thing, but it’s worth circulating in the schema community first. If 20 specs each define the term “agent” slightly differently, what are the implications for search?

**Comment by gkellogg** [comment]
@marcoscaceres `description` is also difficult to get uniformly, as there is no set style for doing definitions; may be the containing paragraph or sentence, or could be from the dd in a definition list, or anything else.

**Comment by marcoscaceres** [comment]
Yeah, I was thinking the same thing. We might need to establish some kind of explicit opt-in for export purposes (similar to what BikeShed does, but with qualifiers).

```HTML
<!-- "export the parent container's text content --> 
<p>
   The <dfn data-export>Foo</dfn> is a bar. 
</p>

<!-- grab just this sentence --> 
<p>
   Foos are really neat! The <dfn data-export="sentence">Foo</dfn> is a bar. Anyway, bla bla. 
</p>
```
 
We could even have `data-export-closest`, which could take a CSS Selector for complex cases.

**Comment by marcoscaceres** [comment]
Closing this feature request for schema:DefinedTerm SEO support. The proposal was experimental when filed and saw no follow-up in 8 years. If schema.org DefinedTerm becomes widely adopted for specs, this can be revisited.
